lib/Importer/ONNX.cpp

```diff
@@ -427,21 +440,49 @@ void ONNXModelLoader::loadNetwork(onnx::GraphProto &net) {
   /// Load the network operators:
   for (int i = 0; i < net.node_size(); i++) {
     auto &op = net.node(i);
-    loadOperator(op);
+    if (!loadOperator(op)) {
+      unexpectedNodeError(op, "Unsupported operator.");
+    }
   }
 
   assert(net.output_size() && "Network needs external outputs defined.");
   auto *r = getNodeByName(net.output(0).name());
   root_ = G_.createSave("output", r);
 }
 
+std::unique_ptr<ONNXModelLoader> ONNXModelLoader::parse(const void *onnxModel,
+                                                        size_t onnxModelSize,
+                                                        Function &F) {
+  std::unique_ptr<ONNXModelLoader> onnxLoader(new ONNXModelLoader(F));
+
+  onnx::GraphProto modelDef;
+  if (!onnxLoader->loadProto(modelDef, onnxModel, onnxModelSize)) {
+    return nullptr;
+  }
+
+  for (int i = 0; i < modelDef.node_size(); i++) {
```

Comment:
How does this work? Note that we eagerly work with Weights now. We extract them from tensors_ map. Who will populate it for `onnxLoader`?

Reply:
The parsing part is needed in order to verify that certain operators can be loaded.
We do not need to provide the tensor values at this moment.
As long as operators can be loaded and then Glow backend (CPU for example) confirms that it can execute it we should be good.

Reply:
Could you explain what happens upon call to `tensors_` map? Say we have FC in our model. Then `loadFC()` is called. Then it calls `getTensorByName()` to get weights. Which in turn asserts in Debug mode, or continues with `Tensor* = NULL` in Release mode, right?

Reply:
That seems to be correct without explicit initializers this would fail. Looks like we need to simulate those tensors artificially. (Note, initializers are not provided as part of the ONNXIFI call).

Surgical intervention into loader was not successful!